File: java/src/test/java/AdapterTest.java
import adapter.Currency;
import adapter.ExchangeRateV1Adapter;
import adapter.repository.ExchangeRateV1Repository;
import adapter.repository.ExchangeRateV2Repository;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class AdapterTest {
    @Test
    void testAdapter() {
        final var exchangeRateV1 = new ExchangeRateV1Repository();
        final var exchangeRateV2 = new ExchangeRateV2Repository();
        final var exchangeRateV1Adapter = new ExchangeRateV1Adapter(exchangeRateV1);

        final var v1USDToKRW = exchangeRateV1.fromUSD(Currency.KRW);
        assertEquals(1271.14, v1USDToKRW);

        final var v2USDToKRW = exchangeRateV2.getExchangeRate(Currency.USD, Currency.KRW);
        assertEquals(1271.14, v2USDToKRW);

        final var adapterUSDToKRW = exchangeRateV1Adapter.getExchangeRate(Currency.USD, Currency.KRW);
        assertEquals(1271.14, adapterUSDToKRW);


        final var v1KRWToUSD = exchangeRateV1.fromKRW(Currency.USD);
        assertEquals(0.00079, v1KRWToUSD);

        final var v2KRWToUSD = exchangeRateV2.getExchangeRate(Currency.KRW, Currency.USD);
        assertEquals(0.00079, v2KRWToUSD);

        final var adapterKRWToUSD = exchangeRateV1Adapter.getExchangeRate(Currency.KRW, Currency.USD);
        assertEquals(0.00079, adapterKRWToUSD);
    }
}

File: src/test/kotlin/AdapterTest.kt
import adapter.Currency
import adapter.ExchangeRateV1Adapter
import adapter.repository.ExchangeRateV1Repository
import adapter.repository.ExchangeRateV2Repository
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class AdapterTest : FunSpec({
    test("Adapter Run") {
        val exchangeRateV1 = ExchangeRateV1Repository()
        val exchangeRateV2 = ExchangeRateV2Repository()
        val exchangeRateV1Adapter = ExchangeRateV1Adapter(exchangeRateV1)

        val v1USDtoKRW = exchangeRateV1.fromUSD(Currency.KRW)
        v1USDtoKRW shouldBe 1271.14

        val v2USDtoKRW = exchangeRateV2.getExchangeRate(Currency.USD, Currency.KRW)
        v2USDtoKRW shouldBe 1271.14

        val adapterUSDtoKRW = exchangeRateV1Adapter.getExchangeRate(Currency.USD, Currency.KRW)
        adapterUSDtoKRW shouldBe 1271.14


        val v1KRWtoUSD = exchangeRateV1.fromKRW(Currency.USD)
        v1KRWtoUSD shouldBe 0.00079

        val v2KRWtoUSD = exchangeRateV2.getExchangeRate(Currency.KRW, Currency.USD)
        v2KRWtoUSD shouldBe 0.00079

        val adapterKRWtoUSD = exchangeRateV1Adapter.getExchangeRate(Currency.KRW, Currency.USD)
        adapterKRWtoUSD shouldBe 0.00079
    }
})